src/Microsoft.Health.Fhir.Core/Models/ResourceChangeData.cs

```diff
@@ -0,0 +1,37 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Fhir.Core.Models
+{
+    public class ResourceChangeData
+    {
+        public ResourceChangeData(long id, DateTime timestamp, string resourceId, short resourceTypeId, int resourceVersion, byte resourceChangeTypeId, string resourceTypeName)
+        {
+            Id = id;
+            Timestamp = timestamp;
+            ResourceId = resourceId;
+            ResourceTypeId = resourceTypeId;
+            ResourceVersion = resourceVersion;
+            ResourceChangeTypeId = resourceChangeTypeId;
+            ResourceTypeName = resourceTypeName;
+        }
+
+        public long Id { get; }
+
+        public string ResourceId { get; }
+
+        public short ResourceTypeId { get; }
+
+        public int ResourceVersion { get; }
+
+        public byte ResourceChangeTypeId { get; }
```

Comment:
I understand that this is efficiently stored as a `tinyint` in the `dbo.ResourceChangeData` table. Have you considered using an enum instead? It would be less space efficient but more readable.

We made this trade off in past with the `OperationStatus` enum used for the `JobRecord.Status` property that is stored as `varchar(10)` in the `dbo.ExportJob` table.

Reply:
There is a separate table called ResourceChangeType for resource change types due to support ad-hoc queries.

Reply:
As discussed, there will be lot of rows in the `dbo.ResourceChangeData` table, so small space efficiencies will go a long way here. Good to keep as is!